<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.247">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>fig project - | Data Analysis (RAD-seq) - Ficus |</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">fig project</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./summary.html">
 <span class="menu-text">| The project |</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./figanalysis.html" aria-current="page">
 <span class="menu-text">| Data Analysis (RAD-seq) - <em>Ficus</em> |</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./figanalysistext.html">
 <span class="menu-text">| Comments |</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#some-notes-to-begin-with" id="toc-some-notes-to-begin-with" class="nav-link active" data-scroll-target="#some-notes-to-begin-with">Some notes to begin with</a></li>
  <li><a href="#the-environment" id="toc-the-environment" class="nav-link" data-scroll-target="#the-environment">1) The environment</a></li>
  <li><a href="#the-data" id="toc-the-data" class="nav-link" data-scroll-target="#the-data">2) The data</a></li>
  <li><a href="#quality-control-of-the-data" id="toc-quality-control-of-the-data" class="nav-link" data-scroll-target="#quality-control-of-the-data">3) Quality control of the data</a>
  <ul class="collapse">
  <li><a href="#installing-fastqc" id="toc-installing-fastqc" class="nav-link" data-scroll-target="#installing-fastqc">3.1) Installing fastqc</a></li>
  <li><a href="#running-fastqc" id="toc-running-fastqc" class="nav-link" data-scroll-target="#running-fastqc">3.2) Running fastqc</a></li>
  </ul></li>
  <li><a href="#merging-read-files-from-the-same-sample" id="toc-merging-read-files-from-the-same-sample" class="nav-link" data-scroll-target="#merging-read-files-from-the-same-sample">4) Merging read files from the same sample</a></li>
  <li><a href="#mapping-rad-reads-on-the-reference-genome" id="toc-mapping-rad-reads-on-the-reference-genome" class="nav-link" data-scroll-target="#mapping-rad-reads-on-the-reference-genome">5) Mapping RAD reads on the reference genome</a>
  <ul class="collapse">
  <li><a href="#installing-bowtie2" id="toc-installing-bowtie2" class="nav-link" data-scroll-target="#installing-bowtie2">5.1) Installing bowtie2</a></li>
  <li><a href="#downloading-the-reference-genome" id="toc-downloading-the-reference-genome" class="nav-link" data-scroll-target="#downloading-the-reference-genome">5.2) Downloading the reference genome</a></li>
  <li><a href="#indexing-the-reference-genome" id="toc-indexing-the-reference-genome" class="nav-link" data-scroll-target="#indexing-the-reference-genome">5.3) Indexing the reference genome</a></li>
  <li><a href="#running-bowtie2" id="toc-running-bowtie2" class="nav-link" data-scroll-target="#running-bowtie2">5.4) Running bowtie2</a></li>
  <li><a href="#install-samtools" id="toc-install-samtools" class="nav-link" data-scroll-target="#install-samtools">5.5) Install samtools</a></li>
  <li><a href="#convert-.sam-to-.bam-and-index-.bam-files" id="toc-convert-.sam-to-.bam-and-index-.bam-files" class="nav-link" data-scroll-target="#convert-.sam-to-.bam-and-index-.bam-files">5.6) Convert .sam to .bam and index .bam files</a></li>
  </ul></li>
  <li><a href="#snp-calling-and-fstatistics" id="toc-snp-calling-and-fstatistics" class="nav-link" data-scroll-target="#snp-calling-and-fstatistics">6) SNP calling and Fstatistics</a>
  <ul class="collapse">
  <li><a href="#installing-stacks" id="toc-installing-stacks" class="nav-link" data-scroll-target="#installing-stacks">6.1) Installing Stacks</a></li>
  <li><a href="#preparing-the-environment" id="toc-preparing-the-environment" class="nav-link" data-scroll-target="#preparing-the-environment">6.2) Preparing the environment</a></li>
  <li><a href="#running-stacks---popmap" id="toc-running-stacks---popmap" class="nav-link" data-scroll-target="#running-stacks---popmap">6.3) Running stacks - popmap</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">| Data Analysis (RAD-seq) - <em>Ficus</em> |</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="some-notes-to-begin-with" class="level2">
<h2 class="anchored" data-anchor-id="some-notes-to-begin-with">Some notes to begin with</h2>
<ul>
<li>You might want to edit the installation scripts to fit</li>
</ul>
<ol type="i">
<li>your system (Linux vs Mac)</li>
<li>the version of the software (latest for August 2022 documented here).</li>
</ol>
<ul>
<li>Depending on your system, you can run a script using (for example):</li>
</ul>
<p><code>sbatch my_script.sh</code> (if your HPC system is using the slurm workload manager). To check status of your job, run (for example): <code>squeue --user=yourname --long</code></p>
<p><code>sh my_script.sh</code> (for a usual script run)</p>
<ul>
<li>If you are working on a Mac OS, use this command when downloading through url links instead of the wget command:</li>
</ul>
<p><code>curl -O url.link</code></p>
<ul>
<li><p>If you have the programs already installed, feel free to adjust your code to simplify the script.</p></li>
<li><p>To ensure your scripts are executable, change the permissions by running: <code>chmod a+x script.sh</code></p></li>
<li><p>Always check the terminal output (or slurm-job.out file)!</p></li>
<li><p>Sadly, Stacks is very tricky to install in a Mac. You might want to consider doing the analysis in a Linux environment instead.</p></li>
</ul>
</section>
<section id="the-environment" class="level2">
<h2 class="anchored" data-anchor-id="the-environment">1) The environment</h2>
<p>To ensure this pipeline will work on your system, you need to have the same environment setup as the one described here. For a start, in your current folder = directory (I called this <strong>fig_analysis</strong>) you need:</p>
<ol type="i">
<li>one folder with your raw data (fastq.gz), named <strong>raw_data</strong>. You can create this directory by running:</li>
</ol>
<p><code>mkdir raw_data</code></p>
<p>Then, upload/transfer your data in this directory. You can do this by running (for example in my case):</p>
<p><code>scp 784*fastq.gz sotiriaboutsi@cyverse.warwick.ac.uk:/home/sotiriaboutsi/fig_analysis/raw_data</code></p>
<ol start="2" type="i">
<li>one folder with the required software (we will install as we go). You can create this directory by running:</li>
</ol>
<p><code>mkdir software</code></p>
</section>
<section id="the-data" class="level2">
<h2 class="anchored" data-anchor-id="the-data">2) The data</h2>
<p>Our data are single-end RAD-seq reads, demultiplexed by SNPSaurus (separate files for each run per sample) and have a single index barcode (no inline barcode in sequence, no need to trim).</p>
<p>Each sample has reads from multiple runs. To view first 10 lines of the data, you can run:</p>
<p><code>zcat ./raw_data/filename.gz | head -10</code></p>
<p>Keep in mind that this will not work in a Mac OS operating system.</p>
</section>
<section id="quality-control-of-the-data" class="level2">
<h2 class="anchored" data-anchor-id="quality-control-of-the-data">3) Quality control of the data</h2>
<p>We will run fastqc for the raw reads to get an idea of the sample quality.</p>
<section id="installing-fastqc" class="level3">
<h3 class="anchored" data-anchor-id="installing-fastqc">3.1) Installing fastqc</h3>
<p>To install fastqc, you can run the <a href="./docs/script files/fastqc_intall_linux.sh">fastqc_intall_linux.sh</a> script, <em>or</em> run the following commands:</p>
<p><code>wget  -P ./software https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip --no-check-certificate</code></p>
<p><code>unzip ./software/fastqc_v0.11.9.zip</code> <code>mv FastQC software</code> <code>chmod a+x ./software/FastQC/fastqc</code></p>
</section>
<section id="running-fastqc" class="level3">
<h3 class="anchored" data-anchor-id="running-fastqc">3.2) Running fastqc</h3>
<p>You can now run fastqc in your raw data by running the <a href="./docs/script files/fastqc.sh">fastqc.sh</a> script <em>or</em> by running:</p>
<p><code>for i in ./raw_data/*.fastq.gz</code> <code>do</code> <code>./software/FastQC/fastqc ${i}</code> <code>done</code></p>
<p>You can now make a new folder called <strong>fastqc_results</strong> and move there the results of the fastqc analysis:</p>
<p><code>mkdir fastqc_results</code> <code>mv ./raw_data/*fastqc* ./fastqc_results</code></p>
<p>You can then download the .html files and open for inspection. To download the files to your local machine, you can use (for example):</p>
<p><code>scp sotiriaboutsi@cyverse.warwick.ac.uk:/home/sotiriaboutsi/fig_analysis/fastqc_results/"*"html .</code></p>
<p>To download the entire directory to your local machine, you can use (for example):</p>
<p><code>scp -r sotiriaboutsi@cyverse.warwick.ac.uk:/home/sotiriaboutsi/fig_analysis/fastqc_results .</code></p>
</section>
</section>
<section id="merging-read-files-from-the-same-sample" class="level2">
<h2 class="anchored" data-anchor-id="merging-read-files-from-the-same-sample">4) Merging read files from the same sample</h2>
<p>Since each sample has reads in multiple runs, I merge the raw files to a single file for each sample. To do this, I need to match all the relevant radtags to the right sample based on the key file provided by SNPSaurus (you can have a look at the key file here: <a href="./docs/data files/fig1617_key.txt">fig1617_key.txt</a>) Due to the relative complexity of the matching (radtags are shared across runs) I will do this manually to ensure there are no mistakes, so there is no shortcut or for-loop possible for this step. You can decide to exclude some raw reads file if the quality is bad.</p>
<p>First, I create a folder for the coming merged files:</p>
<p><code>mkdir raw_data_per_sample</code></p>
<p>Then I run the script merging_data.sh (you can find it here:<a href="./docs/script files/merging_data.sh">merging_data.sh</a>) which replicates this type of command for each sample:</p>
<p><code>cat ./raw_data/raw_file_1.fastq.gz ./raw_data/raw_file_2.fastq.gz &gt; ./raw_data_per_sample/sample_name.gz</code></p>
<p>After this step, you might want to free space by removing the raw reads. <strong>Be careful and make sure you do not need them anymore, as uploading again will be a big waste of time!</strong></p>
<p><code>rm -r raw_reads</code></p>
<p>We should now gunzip the compressed files to feed them to the alignment tool. This takes a lot of storage, so ensure you have enough space in your device.</p>
<p><code>gunzip ./raw_data_per_sample/*</code></p>
</section>
<section id="mapping-rad-reads-on-the-reference-genome" class="level2">
<h2 class="anchored" data-anchor-id="mapping-rad-reads-on-the-reference-genome">5) Mapping RAD reads on the reference genome</h2>
<p>Now my raw rad data are ready to be aligned to the reference genome, one sample at the time. First I need to install the alignment tool of my choice. I use bowtie2. (Bowtie2 is already pre-installed in cyverse for me.)</p>
<section id="installing-bowtie2" class="level3">
<h3 class="anchored" data-anchor-id="installing-bowtie2">5.1) Installing bowtie2</h3>
<p>If you need to install bowtie2, you need to run the bowtie2_install_linux.sh script (<a href="./docs/script files/bowtie2_install_linux.sh">bowtie2_install_linux.sh</a>) <em>or</em> the bowtie2_install_mac.sh script (<a href="./docs/script files/bowtie2_install_mac.sh">bowtie2_install_mac.sh</a>) script <em>or</em> run the following commands:</p>
<p><code>wget  -P ./software https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.4.5/bowtie2-2.4.5-linux-x86_64.zip --no-check-certificate</code></p>
<p><code>unzip ./software/bowtie2-2.4.5-linux-x86_64.zip</code> <code>mv bowtie2-2.4.5-linux-x86_64 software</code></p>
</section>
<section id="downloading-the-reference-genome" class="level3">
<h3 class="anchored" data-anchor-id="downloading-the-reference-genome">5.2) Downloading the reference genome</h3>
<p>After installation, you can prepare for the alignment.</p>
<p><code>mkdir reference</code></p>
<p>Download the reference genome (one file for whole genome and one file for cds regions only) by running reference_download.sh script (<a href="./docs/script files/reference_download.sh">reference_download.sh</a>) <em>or</em> by running:</p>
<p><code>wget  -P ./reference https://ftp.cngb.org/pub/CNSA/data3/CNP0000674/CNS0327487/CNA0019284/Fpum.genome.fa.gz --no-check-certificate</code></p>
<p><code>wget  -P ./reference https://ftp.cngb.org/pub/CNSA/data3/CNP0000674/CNS0327487/CNA0019284/Fpum.genome.cds.fa.gz --no-check-certificate</code></p>
</section>
<section id="indexing-the-reference-genome" class="level3">
<h3 class="anchored" data-anchor-id="indexing-the-reference-genome">5.3) Indexing the reference genome</h3>
<p>First step is to index the reference genome. Run reference_index.sh (<a href="./docs/script files/reference_index.sh">reference_index.sh</a>) <em>or</em> reference_index_mac.sh (<a href="./docs/script files/reference_index_mac.sh">reference_index_mac.sh</a>) <em>or</em> run the following commands:</p>
<p><code>./software/bowtie2-2.4.5-linux-x86_64/bowtie2-build -f ./reference/Fpum.genome.fa.gz index_pum</code></p>
<p><code>./software/bowtie2-2.4.5-linux-x86_64/bowtie2-build -f ./reference/Fpum.genome.cds.fa.gz index_cds_pum</code></p>
<p><code>mv index* ./reference</code></p>
<p>Now we are ready to align our short reads to the reference.</p>
</section>
<section id="running-bowtie2" class="level3">
<h3 class="anchored" data-anchor-id="running-bowtie2">5.4) Running bowtie2</h3>
<p>I use default parameters as they seem to work okay for my analysis. First, make a folder for results:</p>
<p><code>mkdir alignment_results</code></p>
<p>Use the alignment_wg.sh (<a href="./docs/script files/alignment_wg.sh">alignment_wg.sh</a>) and alignment_cds.sh (<a href="./docs/script files/alignment_cds.sh">alignment_cds.sh</a>) scripts <em>or</em> run the following commands:</p>
<p><code>for i in ./raw_data_per_sample/*.fastq</code> <code>do</code> <code>echo ${i}</code> <code>./software/bowtie2-2.4.5-linux-x86_64/bowtie2 -x ./reference/index_pum -U ${i} -S ${i}_wg.sam</code> <code>done</code></p>
<p><code>mv ./raw_data_per_sample/*.sam ./alignment_results</code></p>
<p><code>for i in ./raw_data_per_sample/*.fastq</code> <code>do</code> <code>echo ${i}</code> <code>./software/bowtie2-2.4.5-linux-x86_64/bowtie2 -x ./reference/index_cds_pum -U ${i} -S ${i}_cds.sam</code> <code>done</code></p>
<p><code>mv ./raw_data_per_sample/*.sam ./alignment_results</code></p>
<p><strong>! Do NOT delete the slurm-out files from these jobs! Alternatively, save the terminal output: this gives you information on the alignment details.</strong></p>
<p>We will also need to generate the .bam files and index them.</p>
</section>
<section id="install-samtools" class="level3">
<h3 class="anchored" data-anchor-id="install-samtools">5.5) Install samtools</h3>
<p>First we need to install samtools. Run the samtools_install.sh script (<a href="./docs/script files/samtools_install.sh">samtools_install.sh</a>) <em>or</em> run:</p>
<p><code>wget -P ./software https://github.com/samtools/samtools/releases/download/1.3.1/samtools-1.3.1.tar.bz2 -O samtools.tar.bz2</code> <code>tar -xjvf ./software/samtools.tar.bz2</code> <code>cd ./software/samtools-1.3.1</code> <code>make</code> <code>cd ..</code></p>
</section>
<section id="convert-.sam-to-.bam-and-index-.bam-files" class="level3">
<h3 class="anchored" data-anchor-id="convert-.sam-to-.bam-and-index-.bam-files">5.6) Convert .sam to .bam and index .bam files</h3>
<p>Make .bam files and index them by running the samtools.sh (<a href="./docs/script files/samtools.sh">samtools.sh</a>) script, <em>or</em> samtools_mac.sh <a href="./docs/script files/samtools_mac.sh">samtools_mac.sh</a> (if samtools is preinstalled in your local machine in any path) <em>or</em> the following commands:</p>
<p><code>for i in ./alignment_results/*.sam</code> <code>do</code> <code>./software/samtools-1.3.1/samtools sort -o ${i}.bam ${i}</code> <code>done</code></p>
<p><code>for i in ./alignment_results/*.bam</code> <code>do</code> <code>./software/samtools-1.3.1/samtools index ${i}</code> <code>done</code></p>
</section>
</section>
<section id="snp-calling-and-fstatistics" class="level2">
<h2 class="anchored" data-anchor-id="snp-calling-and-fstatistics">6) SNP calling and Fstatistics</h2>
<p>Once the raw data are mapped onto a reference genome, we can run the stacks pipeline. We are using ref_map.pl since our reads have been aligned based on a reference genome.</p>
<section id="installing-stacks" class="level3">
<h3 class="anchored" data-anchor-id="installing-stacks">6.1) Installing Stacks</h3>
<p>First we need to install stacks (you need to adjust to your directory/system).</p>
<p><em>This is very tricky to do in a Mac, I haven’t figured out how to install it properly! I did my analysis in the cyverse.</em></p>
<p>Run:</p>
<p><code>wget  -P ./software http://creskolab.uoregon.edu/stacks/source/stacks-2.62.tar.gz --no-check-certificate</code> <code>tar -vxf ./software/stacks-2.62.tar.gz</code> <code>mv stacks-2.62 software</code> <code>cd software/stacks-2.62</code> <code>scl enable devtoolset-8 bash</code> <code>./configure --prefix /home/sotiriaboutsi</code> <code>make</code> <code>make install</code> <code>cd ..</code> <code>cd ..</code></p>
</section>
<section id="preparing-the-environment" class="level3">
<h3 class="anchored" data-anchor-id="preparing-the-environment">6.2) Preparing the environment</h3>
<p>To run the analysis, you need to upload the “population map” .txt files in the <em>fig_analysis</em> directory. These files will match the sample names to their assigned population/species. The number (3 or 4) indicates whether data from <em>F.trichocerasa</em> have been included (useful as an outgroup).</p>
<p>The files:</p>
<ol type="i">
<li><p>whole genome alignments: <a href="./docs/data files/wg_3_populations.txt">wg_3_populations.txt</a> <a href="./docs/data files/wg_4_populations.txt">wg_4_populations.txt</a> <a href="./docs/data files/wg_3_species.txt">wg_3_species.txt</a> <a href="./docs/data files/wg_4_species.txt">wg_4_species.txt</a></p></li>
<li><p>cds alignments: <a href="./docs/data files/cds_3_populations.txt">cds_3_populations.txt</a> <a href="./docs/data files/cds_4_populations.txt">cds_4_populations.txt</a> <a href="./docs/data files/cds_3_species.txt">cds_3_species.txt</a> <a href="./docs/data files/cds_4_species.txt">cds_4_species.txt</a></p></li>
</ol>
</section>
<section id="running-stacks---popmap" class="level3">
<h3 class="anchored" data-anchor-id="running-stacks---popmap">6.3) Running stacks - popmap</h3>
<p>Now we can run the ref_map.pl program script from stacks on our data by the running the “popmap_reference_speciesnumber_analysislevel.sh” scripts.</p>
<section id="stacks-for-wg---3-species" class="level4">
<h4 class="anchored" data-anchor-id="stacks-for-wg---3-species">6.3.1) stacks for wg - 3 species</h4>
<p>We with start with the data that have been mapped on the whole genome (wg) and for the subset of the 3 focal species, both at the species and t the population level.</p>
<p>The scripts are available here: <a href="./docs/script files/popmap_wg_3_species.sh">popmap_wg_3_species.sh</a> <a href="./docs/script files/popmap_wg_3_populations.sh">popmap_wg_3_populations.sh</a></p>
<p>You can also do this by running the following commands:</p>
<p><code>ref_map.pl --samples ./alignment_results --popmap wg_3_species.txt -o ./popmap_results/wg_3_species -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap wg_3_species.txt -o ./popmap_results/wg_3_species -X "populations:--vcf"</code></p>
<p><code>ref_map.pl --samples ./alignment_results --popmap wg_3_populations.txt -o ./popmap_results/wg_3_populations -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap wg_3_populations.txt -o ./popmap_results/wg_3_populations -X "populations:--vcf"</code></p>
<p>Then we can continue with the rest of the subsets and mapping.</p>
</section>
<section id="stacks-for-wg---4-species" class="level4">
<h4 class="anchored" data-anchor-id="stacks-for-wg---4-species">6.3.2) stacks for wg - 4 species</h4>
<p>The scripts are available here: <a href="./docs/script files/popmap_wg_4_species.sh">popmap_wg_4_species.sh</a> <a href="./docs/script files/popmap_wg_4_populations.sh">popmap_wg_4_populations.sh</a></p>
<p>You can also do this by running the following commands:</p>
<p><code>ref_map.pl --samples ./alignment_results --popmap wg_4_species.txt -o ./popmap_results/wg_4_species -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap wg_4_species.txt -o ./popmap_results/wg_4_species -X "populations:--vcf"</code></p>
<p><code>ref_map.pl --samples ./alignment_results --popmap wg_4_populations.txt -o ./popmap_results/wg_4_populations -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap wg_4_populations.txt -o ./popmap_results/wg_4_populations -X "populations:--vcf"</code></p>
</section>
<section id="stacks-for-cds---3-species" class="level4">
<h4 class="anchored" data-anchor-id="stacks-for-cds---3-species">6.3.3) stacks for cds - 3 species</h4>
<p>The scripts are available here: <a href="./docs/script files/popmap_cds_3_species.sh">popmap_cds_3_species.sh</a> <a href="./docs/script files/popmap_cds_3_populations.sh">popmap_cds_3_populations.sh</a></p>
<p>You can also do this by running the following commands:</p>
<p><code>ref_map.pl --samples ./alignment_results --popmap cds_3_species.txt -o ./popmap_results/cds_3_species -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap cds_3_species.txt -o ./popmap_results/cds_3_species -X "populations:--vcf"</code></p>
<p><code>ref_map.pl --samples ./alignment_results --popmap cds_3_populations.txt -o ./popmap_results/cds_3_populations -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap cds_3_populations.txt -o ./popmap_results/cds_3_populations -X "populations:--vcf"</code></p>
</section>
<section id="stacks-for-cds---4-species" class="level4">
<h4 class="anchored" data-anchor-id="stacks-for-cds---4-species">6.3.4) stacks for cds - 4 species</h4>
<p>The scripts are available here: <a href="./docs/script files/popmap_cds_4_species.sh">popmap_cds_4_species.sh</a> <a href="./docs/script files/popmap_cds_4_populations.sh">popmap_cds_4_populations.sh</a></p>
<p>You can also do this by running the following commands:</p>
<p><code>ref_map.pl --samples ./alignment_results --popmap cds_4_species.txt -o ./popmap_results/cds_4_species -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap cds_4_species.txt -o ./popmap_results/cds_4_species -X "populations:--vcf"</code></p>
<p><code>ref_map.pl --samples ./alignment_results --popmap cds_4_populations.txt -o ./popmap_results/cds_4_populations -X "populations:--fstats"</code> <code>ref_map.pl --samples ./alignment_results --popmap cds_4_populations.txt -o ./popmap_results/cds_4_populations -X "populations:--vcf"</code></p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>